pikaur/pprint.py

```diff
@@ -68,41 +69,19 @@ def pretty_format_upgradeable(packages_updates, verbose=False, print_repo=False,
     if not color:
         _color_line = _bold_line = lambda x, *args: x
 
-    def get_common_string(str1, str2):
-        result = ''
-        if '' in (str1, str2):
-            return result
-        counter = 0
-        while (
-                counter < len(str1)
-        ) and (
-            counter < len(str2)
-        ) and (
-            str1[counter] == str2[counter]
-        ):
-            result += str1[counter]
-            counter += 1
-        return result
-
-    def get_version_diff(version, common_version):
-        new_version_postfix = version
-        if common_version != '':
-            _new_version_postfix = version.split(
-                common_version
-            )[1:]
-            new_version_postfix = common_version.join(_new_version_postfix)
-        return new_version_postfix
-
     def pretty_format(pkg_update):
-        common_version = get_common_string(
-            pkg_update.Current_Version, pkg_update.New_Version
-        )
+        current_version = PackageVersion(pkg_update.Current_Version)
+        new_version = PackageVersion(pkg_update.New_Version)
+
+        _, prefix1 = current_version.compare(new_version)
+        _, prefix2 = new_version.compare(current_version)
+
         version_color = 10
         old_color = 11
         new_color = 9
         column_width = min(int(get_term_width() / 2.5), 37)
         sort_by = '{:03d}{}'.format(
-            len(common_version)*10+len(pkg_update.New_Version),
+            prefix1*10+len(pkg_update.New_Version),
```

Comment:
you should sort not by length of common string but by number of blocks

i think it will be better to completely rewrite it all again and and just replace `get_common_string` and `get_version_diff` to one function `get_common_version(version1, version2)` which will return common version part (as string) and length of common blocks